### PR TITLE
Add clcache for windows

### PR DESCRIPTION
**Pull request by OwenArnold**
WIP, see upstream pipelines https://github.com/scipp/pipelines/pull/8

### CODE REVIEW

**Comment by SimonHeybrock** [comment]
@OwenArnold Conflicts need resolving before we can merge.

File: tools/build_cpp.py [hunk 1]
```diff
@@ -100,14 +106,20 @@ def main(prefix='install', build_dir='build', source_dir='.'):
     run_command(['cmake', '-B', '.', '-S', source_dir, '-LA'], shell=shell)
 
     # Compile benchmarks, C++ tests, and python library
+    start = time.time()
```
Comment (nvaytet) [inline tools/build_cpp.py:109]: Do we want to keep the timings or was this for debugging?

Comment (OwenArnold) [reply tools/build_cpp.py:109]: It was originally for debugging, but we should keep I think to see how we are getting on with ccache and compilation times.

File: scippneutron-developer-no-mantid.yml [hunk 2]
```diff
@@ -17,6 +17,8 @@ dependencies:
   - ninja
   - tbb-devel
   - scipp
+  - cmake
+  - pip
```
Comment (nvaytet) [inline scippneutron-developer-no-mantid.yml:21]: Why do you need `pip` here?

Comment (OwenArnold) [reply scippneutron-developer-no-mantid.yml:21]: This is where we get ccache from.

Comment (nvaytet) [reply scippneutron-developer-no-mantid.yml:21]: And the cmake is because the system version is too old?

Comment (OwenArnold) [reply scippneutron-developer-no-mantid.yml:21]: Yes, we've also had some pain from this before I think (out of date system cmake), better we control in our environments and have cross-platform consistency.

**Review by nvaytet — APPROVED** [review]

**Pull request merged**